src/InsetOrderOptimizer.cpp

```diff
@@ -159,142 +145,175 @@ bool InsetOrderOptimizer::addToLayer()
     return added_something;
 }
 
-
-std::unordered_set<std::pair<const ExtrusionLine*, const ExtrusionLine*>> InsetOrderOptimizer::getRegionOrder(const std::vector<const ExtrusionLine*>& input, const bool outer_to_inner)
+InsetOrderOptimizer::value_type InsetOrderOptimizer::getRegionOrder(const auto& input, const bool outer_to_inner)
 {
-    std::unordered_set<std::pair<const ExtrusionLine*, const ExtrusionLine*>> order_requirements;
-
-    // We build a grid where we map toolpath vertex locations to toolpaths,
-    // so that we can easily find which two toolpaths are next to each other,
-    // which is the requirement for there to be an order constraint.
-    //
-    // We use a PointGrid rather than a LineGrid to save on computation time.
-    // In very rare cases two insets might lie next to each other without having neighboring vertices, e.g.
-    //  \            .
-    //   |  /        .
-    //   | /         .
-    //   ||          .
-    //   | \         .
-    //   |  \        .
-    //  /            .
-    // However, because of how Arachne works this will likely never be the case for two consecutive insets.
-    // On the other hand one could imagine that two consecutive insets of a very large circle
-    // could be simplify()ed such that the remaining vertices of the two insets don't align.
-    // In those cases the order requirement is not captured,
-    // which means that the PathOrderOptimizer *might* result in a violation of the user set path order.
-    // This problem is expected to be not so severe and happen very sparsely.
-
-    coord_t max_line_w = 0u;
-    for (const ExtrusionLine* line : input)
-    { // compute max_line_w
-        for (const ExtrusionJunction& junction : *line)
-        {
-            max_line_w = std::max(max_line_w, junction.w);
-        }
+    if (input.empty()) // Early out
+    {
+        return {};
     }
-    if (max_line_w == 0u)
-        return order_requirements;
 
-    struct LineLoc
+    // Cache the polygons and get the signed area of each extrusion line and store them mapped against the pointers for those lines
+    struct Locator
```

Comment:
Rename struct to `LineLoc` to minimize the diff
```suggestion
    struct LineLoc
```